unlock-app/src/__tests__/services/web3Service.test.js

```diff
@@ -473,55 +472,138 @@ describe('Web3Service', () => {
+            blockHash: 'null',
+            transactionIndex: null, // Not mined
+            from: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+            to: '0xcfeb869f69431e42cdb54a4f4f105c19c080a601',
+            value: '0x0',
+            gas: '0x16e360',
+            gasPrice: '0x04a817c800',
+            input:
+              '0x2bc888bf00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000278d00000000000000000000000000000000000000000000000000002386f26fc10000000000000000000000000000000000000000000000000000000000000000000a',
+          })
+
+          web3Service._watchTransaction = jest.fn()
+          web3Service.getTransactionType = jest.fn(() => 'TRANSACTION_TYPE')
         })
 
-        return web3Service.getTransaction(transaction.hash)
+        it('should watch the transaction', done => {
+          expect.assertions(1)
+          web3Service.on('transaction.updated', () => {
+            expect(web3Service._watchTransaction).toHaveBeenCalledWith(
+              transaction.hash
+            )
+            done()
+          })
+
+          web3Service.getTransaction(transaction.hash)
+        })
+
+        it('should emit a transaction.updated event with the right values', done => {
+          expect.assertions(5)
+          web3Service.on('transaction.updated', (hash, update) => {
+            expect(hash).toBe(transaction.hash)
+            expect(update.status).toEqual('pending')
+            expect(update.confirmations).toEqual(0)
+            expect(update.blockNumber).toEqual(Number.MAX_SAFE_INTEGER)
+            expect(update.type).toEqual('TRANSACTION_TYPE')
+            done()
+          })
+
+          web3Service.getTransaction(transaction.hash)
+        })
       })
 
-      it('should trigger and error if the transaction could not be found', done => {
-        expect.assertions(1)
+      describe('when the transaction has been mined but not fully confirmed', () => {
+        beforeEach(() => {
+          ethBlockNumber(`0x${(17).toString('16')}`)
+          ethGetTransactionByHash(transaction.hash, {
+            hash:
+              '0x83f3e76db42dfd5ebba894e6ff462b3ae30b5f7bfb7a6fec3888e0ed88377f64',
+            nonce: '0x04',
+            blockHash:
+              '0xdc7c95899e030f3104871a597866767ec296e948a24e99b12e0b251011d11c36',
+            blockNumber: `0x${(14).toString('16')}`,
+            transactionIndex: '0x0d',
+            from: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+            to: '0xcfeb869f69431e42cdb54a4f4f105c19c080a601',
+            value: '0x0',
+            gas: '0x16e360',
+            gasPrice: '0x04a817c800',
+            input:
+              '0x2bc888bf00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000278d00000000000000000000000000000000000000000000000000002386f26fc10000000000000000000000000000000000000000000000000000000000000000000a',
```

Comment:
Would you be able to adjust the formatting of this input value?

Reply:
Let me try but I am actually not sure how this is built :/ (it was taken from a nock spy)

Reply:
huh, this is a mess :/ I'll push that to another PR if you don't mind.